### PR TITLE
BUGFIX: Fallback to jpg if webp is not supported on login screen

### PR DESCRIPTION
**Pull request by jonnitto**
This is the following PR for #3413

If the browser doesn't support `webp`, it will fall back to the same filename with `jpg`.

### CODE REVIEW

**Comment by jonnitto** [comment]
@daniellienert Could you add a `jpg` variant of the login wallpaper?

**Comment by daniellienert** [comment]
@jonnitto : done

**Comment by markusguenther** [comment]
Hi I tried this PR now on neos72.unikka.de and sadly it does not work. On the webp supported system just the loginbox background works and on the non supported system it is still black.

I used Mojave with safari 14 now.

**Comment by markusguenther** [comment]
Need to test it locally for a better understanding. Just used the neos72.unikka.de to not install everything on the older mac.

**Comment by kdambekalns** [comment]
Tried neos72.unikka.de with Firefox on Big Sur, and that's broken, too:

image

**Comment by markusguenther** [comment]
Ok seems that the  detection does not work properly. Locally, the script always serves no-webp for me.
Need more investigation.

**Comment by markusguenther** [comment]
Ok one step further... the error in Karstens screen comes from an missing jQuery. We fixed that in this PR.
But we need to adjust the Fluid Templates as well ;)

https://github.com/neos/neos-development-collection/pull/3352

**Comment by markusguenther** [comment]
So now I know whats going wrong, hope that @jonnitto or I find the time to fix it soon.

In the regular login template (Fusion) we make the detection for webp compatibilty. An then add the `no-webp` class for instance. This step is missing for the Welcome Error Teamplate (Fluid). So the Fluid template has no global background image defined, but the regular login has.

Sadly the webp detection also does not work on my machine, but as we have the fallback it does not look broken.
So we need a solution for this part for the fluid template.

```
                <script @if.set={props.backgroundImageSourceIsWebp}>
                    {"(function(){var webp = false, elem = document.createElement('canvas'); if (!!(elem.getContext && elem.getContext('2d'))) { webp = elem.toDataURL('image/webp').indexOf('data:image/webp') == 0 } document.documentElement.classList.add(webp ? 'webp' :'no-webp'); console.log(webp)})()"}
                </script>
                <style type="text/css" @if.set={props.backgroundImageSource && !props.backgroundImageSourceIsWebp}>
                    {'body.neos--bg,.neos-login-box:before{background-image:url(' + props.backgroundImageSource + ')}'}
                </style>
                <style type="text/css" @if.set={props.backgroundImageSource && props.backgroundImageSourceIsWebp}>
                    {'.webp body.neos--bg,.webp .neos-login-box:before {background-image: url(' + props.backgroundImageSource + ')}'}
                    {'.no-webp body.neos--bg,.no-webp .neos-login-box:before {background-image: url(' + String.substr(props.backgroundImageSource, 0, String.length(props.backgroundImageSource) - 4) + 'jpg)}'}
                </style>
```

Then the view will not be broken anymore. Still need to check what's up with the webp check.

**Comment by kdambekalns** [comment]
On the webp check: I conclude this way of checking only works on Chrome.

On https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL it says:

> Chrome also supports the image/webp type.

Since on Firefox for me the `toDataURL` call returns `data:image/png…`, it seems no to work there, even though WebP is supported. Changed it to a different method…

**Review by kdambekalns — APPROVED** [review]

**Review by fcool — APPROVED** [review]
Any reason, why we do not add the javascript for the detection to our application javascript but inline it here?
Or "just" to have it as soon as possible and small and easy?

**Comment by kdambekalns** [comment]
I see two reasons:

- it's only needed in those two places
- the JS without the `img` is useless

File: Neos.Neos/Resources/Private/Fusion/Backend/Views/Login.fusion [hunk 1]
```diff
@@ -63,6 +60,7 @@ prototype(Neos.Neos:View.Login) < prototype(Neos.Fusion:Component) {
                 </style>
             </head>
             <body class={["neos", props.backgroundImageSource ? 'neos--bg' : null]}>
+                <img @if.set={props.backgroundImageSourceIsWebp} style="display: none" src="data:image/webp;base64,UklGRiIAAABXRUJQVlA4IBYAAAAwAQCdASoBAAEADsD+JaQAA3AAAAAA" onload="document.documentElement.classList.add('webp')" onerror="document.documentElement.classList.add('no-webp')"/>
```
Comment (jonnitto) [inline Neos.Neos/Resources/Private/Fusion/Backend/Views/Login.fusion:63]: What a nasty trick! Thanks @kdambekalns for taking care ❤️

**Pull request merged**